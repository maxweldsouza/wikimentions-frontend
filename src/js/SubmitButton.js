import React from 'react';

class SubmitButton extends React.Component {
    static get defaultProps () {
        return {
            className: 'button'
        };
    }
    render () {
        const animation = {
            animationName: 'spinner-animation',
            animationDuration: '350ms',
            animationIterationCount: 'infinite',
            animationTimingFunction: 'linear'
        };
        if (this.props.submitting) {
            return <button
                type='submit'
                className={`${this.props.className} loading`}>{this.props.title}
            </button>;
        }
        if (this.props.confirm === false) {
            return <button type='submit' className={`${this.props.className} disabled`} onClick={this.props.onSubmit}>{this.props.title}</button>;
        }
        return <button type='submit' className={this.props.className} onClick={this.props.onSubmit}>{this.props.title}</button>;
    }
}

export default SubmitButton;
