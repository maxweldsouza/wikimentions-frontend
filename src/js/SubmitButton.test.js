import React from 'react';
import renderer from 'react-test-renderer';
import SubmitButton from './SubmitButton';

test('SubmitButton renders correctly', () => {
    const tree = renderer
        .create(<SubmitButton title='Fb' submitting confirm />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
