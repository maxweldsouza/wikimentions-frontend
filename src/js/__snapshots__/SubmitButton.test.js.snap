// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
<button
  className="button loading"
  type="submit"
>
  Fb
</button>
`;
